src/context/Favorite/FavoriteContext.tsx

```diff
@@ -0,0 +1,47 @@
+import { Dish } from '../../types/dishType.ts';
+import { createContext, FC, PropsWithChildren, useContext, useEffect, useState } from 'react';
+import { useGetFavorite } from '../../hooks/useFavoriteDishes/useGetFavorite.ts';
+import { checkTheFavoriteDish } from '../../utils/checkFavoriteDish/checkTheFavoriteDish.ts';
+
+type FavoriteContext = {
+  favorite: Dish[] | [];
+  setFavorite: (data: Dish[]) => void;
+  isFavorite: (data: Dish) => boolean;
+};
+
+const Context = createContext<FavoriteContext>({
+  favorite: [],
+  setFavorite: () => {},
+  isFavorite: () => false,
+});
+
+const useFavoriteContext = () => useContext(Context);
+
+const FavoriteProvider: FC<PropsWithChildren> = ({ children }) => {
```

Comment:
better to keep only IDs in the local storage, sometimes some details can be changed on the external resource and data from local storage will be outdated